src/main/java/net/minecraftforge/resource/PathResourcePack.java

```diff
@@ -108,6 +126,10 @@ public Collection<ResourceLocation> getResources(PackType type, String resourceN
             Path root = resolve(type.getDirectory(), resourceNamespace).toAbsolutePath();
             Path inputPath = root.getFileSystem().getPath(pathIn);
 
+            if (ResourceCacheManager.shouldUseCache() && this.cacheManager.hasCached(type, resourceNamespace)) {
```

Comment:
That will cause confusion because the contexts of both methods are different. One is a static call that validates that the caching is even allowed, while the other only indicates if the manager holds on to the cache value currently.